### PR TITLE
[quant][graphmode] Test JIT tracing for dynamic quant cases

### PR DESCRIPTION
**Pull request by supriyar**
Stack from [ghstack](https://github.com/ezyang/ghstack):
* **#40040 [quant][graphmode] Test JIT tracing for dynamic quant cases**
* #40039 [quant][graphmode] Refactor dynamic quant tests

Summary:

Test Plan:
python test/test_quantization.py TestQuantizeDynamicScriptJitPasses

Reviewers:

Subscribers:

Tasks:

Tags:

Differential Revision: [D22071277](https://our.internmc.facebook.com/intern/diff/D22071277)

### CODE REVIEW

File: test/quantization/test_quantize_script.py [hunk 1]
```diff
@@ -2461,14 +2461,19 @@ def __init__(self):
...
-        FileCheck().check("aten::add") \
-                   .check("quantized::linear_dynamic") \
-                   .run(model.graph)
+        eager_model = M().eval()
+        for tracing_mode in [True, False]:
+            if tracing_mode:
+                x = torch.randn(5, 5)
+                model = torch.jit.trace(eager_model, x)
+            else:
+                model = torch.jit.script(eager_model)
```
Comment (jerryzh168) [inline test/quantization/test_quantize_script.py:2470]: we have a utility function for this: `get_script_module`. please take a look at other tests in the file.

File: test/quantization/test_quantize_script.py [hunk 2]
```diff
@@ -2461,14 +2461,19 @@ def __init__(self):
...
-
-        m = torch.jit.script(M())
-        qconfig_dict = {'' : default_dynamic_qconfig}
-        model = quantize_dynamic_script(m, qconfig_dict)
-        # add op is not dynamically quantized.
-        FileCheck().check("aten::add") \
-                   .check("quantized::linear_dynamic") \
-                   .run(model.graph)
+        eager_model = M().eval()
+        for tracing_mode in [True, False]:
```
Comment (jerryzh168) [inline test/quantization/test_quantize_script.py:2465]: nit: suggest to use `tracing` for the name to align with other tests in the file

File: test/quantization/test_quantize_script.py [hunk 3]
```diff
@@ -2461,14 +2461,16 @@ def __init__(self):
...
-                   .check("quantized::linear_dynamic") \
-                   .run(model.graph)
+        eager_model = M().eval()
+        x = torch.randn(5, 5)
+        for tracing in [True, False]:
+            model = get_script_module(eager_model, tracing, x)
+            qconfig_dict = {'' : default_dynamic_qconfig}
+            model = quantize_dynamic_script(model, qconfig_dict)
+            # add op is not dynamically quantized.
+            FileCheck().check("aten::add") \
```
Comment (jerryzh168) [inline test/quantization/test_quantize_script.py:2471]: can you use `_test_op_impl`? It is in TestQuantizeScriptPTSQOps, but you can move it to QuantizationTestCase

Comment (supriyar) [reply test/quantization/test_quantize_script.py:2471]: Sure. That currently only tests `quantize_script` API. I'll make it generic.

Comment (jerryzh168) [reply test/quantization/test_quantize_script.py:2471]: or maybe define a `_test_op_impl` locally for this test

Comment (supriyar) [reply test/quantization/test_quantize_script.py]: @jerryzh168 refactored it a bit to make it usable by both.

File: test/quantization/test_quantize_script.py [hunk 4]
```diff
@@ -2462,13 +2430,13 @@ def forward(self, x):
...
                 return self.fc1(x)
 
-        m = torch.jit.script(M())
-        qconfig_dict = {'' : default_dynamic_qconfig}
-        model = quantize_dynamic_script(m, qconfig_dict)
-        # add op is not dynamically quantized.
-        FileCheck().check("aten::add") \
-                   .check("quantized::linear_dynamic") \
-                   .run(model.graph)
+        eager_model = M().eval()
```
Comment (jerryzh168) [inline test/quantization/test_quantize_script.py:2433]: btw, `eval_mode` is an argument of `checkGraphModeOp`

**Review by jerryzh168 — APPROVED** [review]
LG, thanks!

**Pull request closed without merging**